packages/wellknown-did-verifier/src/types/IWellKnownDidVerifier.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  IDidConfigurationResource,
+  IDomainLinkageValidation,
+  IVerifyCallbackArgs,
+  IVerifyCredentialResult,
+  IResourceValidation
+} from '@sphereon/wellknown-dids-client/dist/types'
+import {
+  IAgentContext,
+  IPluginMethodMap,
+  IResolver
+} from '@veramo/core'
+
+export interface IWellKnownDidVerifier extends IPluginMethodMap {
+  registerSignatureVerification(args: IRegisterSignatureVerificationArgs, context: IRequiredContext): Promise<void>,
+  removeSignatureVerification(args: IRemoveSignatureVerificationArgs, context: IRequiredContext): Promise<boolean>,
+  verifyDomainLinkage(args: IVerifyDomainLinkageArgs, context: IRequiredContext): Promise<IDomainLinkageValidation>,
+  verifyDidConfigurationResource(args: IVerifyDidConfigurationResourceArgs, context: IRequiredContext): Promise<IResourceValidation>
+}
+
+export interface IWellKnownDidVerifierOptionsArgs {
+  signatureVerifications?: Record<string, (args: IVerifyCallbackArgs) => Promise<IVerifyCredentialResult>>
+  onlyVerifyServiceDids?: boolean
+}
+
+export interface IRegisterSignatureVerificationArgs {
+  key: string
+  signatureVerification: (args: IVerifyCallbackArgs) => Promise<IVerifyCredentialResult>
+}
+
+export interface IRemoveSignatureVerificationArgs {
+  key: string
+}
+
+export interface IVerifyDomainLinkageArgs {
+  didUrl: string
```

Comment:
Is there a reason why this isn't called a did? It really expects a DID there instead of a didUrl right?

Reply:
The thought here was that since we need to resolve the DID first and the resolver can have a didUrl. Then from the DID document the only DID's that can be passed on are the ones in the service. For the verifyResource function it is stated as a did

Reply:
A did resolver can have a DID url because it can actually resolve subsections of a DID or in some cases even external resources present in a DID document. In here we only should have a did, not a didUrl. So it really should be did